src/gatsby-node.js

```diff
@@ -0,0 +1,181 @@
+    actions,
+    schemas,
+    createContentDigest,
+    createNodeId,
+    reporter,
+    getCache,
+    getNode,
+    getNodes,
+    store,
+    cache,
+  };
+
+  const existingNodes = getNodes().filter(
+    (n) => n.internal.owner === `gatsby-source-strapi` || n.internal.type === 'File'
+  );
+
+  existingNodes.forEach((n) => touchNode(n));
+
+  const endpoints = getEndpoints(pluginOptions, schemas);
+
+  const lastFetched = await cache.get(LAST_FETCHED_KEY);
+
+  const allResults = await Promise.all(
+    endpoints.map(({ kind, ...config }) => {
+      if (kind === 'singleType') {
+        return fetchEntity(config, ctx);
+      }
+
+      return fetchEntities(config, ctx);
+    })
+  );
+
+  let newOrExistingEntries;
+
+  // Fetch only the updated data between run
+  if (lastFetched) {
+    // Add the updatedAt filter
+    const deltaEndpoints = endpoints.map((endpoint) => {
+      return {
+        ...endpoint,
+        queryParams: {
+          ...endpoint.queryParams,
+          // TODO
+          filters: {
+            updatedAt: { $gt: lastFetched },
+          },
+        },
+      };
+    });
+
+    newOrExistingEntries = await Promise.all(
+      deltaEndpoints.map(({ kind, ...config }) => {
+        if (kind === 'singleType') {
+          return fetchEntity(config, ctx);
+        }
+
+        return fetchEntities(config, ctx);
+      })
+    );
+  }
+
+  const data = newOrExistingEntries || allResults;
+
+  // Build a map of all nodes with the gatsby id and the strapi_id
+  const existingNodesMap = buildMapFromNodes(existingNodes);
+
+  // Build a map of all the parent nodes that should be removed
+  // This should also delete all the created nodes for markdown, relations, dz...
+  // When fetching only one content type and populating its relations it might cause some issues
+  // as the relation nodes will never be deleted
+  // it's best to fetch the content type and its relations separately and to populate
+  // only one level of relation
+  const nodesToRemoveMap = buildNodesToRemoveMap(existingNodesMap, endpoints, allResults);
+
+  // Delete all nodes that should be deleted
+  Object.entries(nodesToRemoveMap).forEach(([nodeName, nodesToDelete]) => {
+    if (nodesToDelete.length) {
+      reporter.info(`Strapi: ${nodeName} deleting ${nodesToDelete.length}`);
+
+      nodesToDelete.forEach(({ id }) => {
+        const node = getNode(id);
+
+        touchNode(node);
+        deleteNode(node);
+      });
+    }
+  });
+
+  let warnOnceForNoSupport = false;
+
+  await cache.set(LAST_FETCHED_KEY, Date.now());
+
+  for (let i = 0; i < endpoints.length; i++) {
+    const { uid } = endpoints[i];
+
+    await downloadMediaFiles(data[i], ctx, uid);
+
+    for (let entity of data[i]) {
+      const nodes = createNodes(entity, ctx, uid);
+
+      await Promise.all(nodes.map((n) => actions.createNode(n)));
+
+      const nodeType = makeParentNodeName(ctx.schemas, uid);
+
+      const mainEntryNode = nodes.find((n) => {
```

Comment:
Is it possible to edit a block outside of a page in Strapi? In Contentful we create node manifests for all content nodes (even if they're not top level) because it means you can press "Open Preview" from within the edit view of a nested block.

Reply:
No it's not possible, only primary nodes, the one that create pages, regarding linked nodes the manifest will also be created for them as we fetch them